OneSignalExample/Assets/OneSignal/Editor/PostProcessBuildPlayer_iOS.cs

```diff
@@ -1,60 +1,140 @@
+
+         Directory.CreateDirectory (pathToNotificationService);
+
+         var plistPath = "Assets/OneSignal/Platforms/iOS/Info.plist";
+
+         var notificationServicePlistPath = path + "/OneSignalNotificationExtensionService/Info.plist";
+
+         PlistDocument notificationServicePlist = new PlistDocument();
+         notificationServicePlist.ReadFromFile (plistPath);
+         notificationServicePlist.root.SetString ("CFBundleShortVersionString", PlayerSettings.bundleVersion);
+         notificationServicePlist.root.SetString ("CFBundleVersion", PlayerSettings.iOS.buildNumber.ToString ());
+
+         var notificationServiceTarget = PBXProjectExtensions.AddAppExtension (project, targetGUID, "OneSignalNotificationExtensionService", PlayerSettings.GetApplicationIdentifier (BuildTargetGroup.iOS) + ".OneSignalNotificationExtensionService", notificationServicePlistPath);
+
+         project.AddFileToBuild (notificationServiceTarget, project.AddFile ("OneSignalNotificationExtensionService/NotificationService.h", "OneSignalNotificationExtensionService/NotificationService.h", PBXSourceTree.Source));
+         project.AddFileToBuild (notificationServiceTarget, project.AddFile ("OneSignalNotificationExtensionService/NotificationService.m", "OneSignalNotificationExtensionService/NotificationService.m", PBXSourceTree.Source));
+
+         project.AddFrameworkToProject (notificationServiceTarget, "NotificationCenter.framework", true);
+         project.AddFrameworkToProject (notificationServiceTarget, "UserNotifications.framework", true);
+         project.AddFrameworkToProject (notificationServiceTarget, "UIKit.framework", true);
+         project.AddFrameworkToProject (notificationServiceTarget, "SystemConfiguration.framework", true);
+
+         project.SetBuildProperty (notificationServiceTarget, "ARCHS", "$(ARCHS_STANDARD)");
+         project.SetBuildProperty (notificationServiceTarget, "DEVELOPMENT_TEAM", PlayerSettings.iOS.appleDeveloperTeamID);
+
+         notificationServicePlist.WriteToFile (notificationServicePlistPath);
+
+         FileUtil.CopyFileOrDirectory ("Assets/OneSignal/Platforms/iOS/NotificationService.h", path + "/OneSignalNotificationExtensionService/NotificationService.h");
+         FileUtil.CopyFileOrDirectory ("Assets/OneSignal/Platforms/iOS/NotificationService.m", path + "/OneSignalNotificationExtensionService/NotificationService.m");
+
+         project.WriteToFile (projPath);
 
-public class PostBuildTrigger {
+         string projectFile = path + "/Unity-iPhone.xcodeproj/project.pbxproj";
+         string contents = File.ReadAllText(projectFile);
 
-   [PostProcessBuild]
-   public static void OnPostProcessBuild(BuildTarget target, string pathToBuiltProject) {
-      LinkLibraries(target, pathToBuiltProject);
+         InsertStaticFrameworkIntoTargetBuildPhaseFrameworks("libOneSignal", "CD84C25F20742FAB0035D524", notificationServiceTarget, ref contents, project);
```

Comment:
Is this `CD84C25F20742FAB0035D524` hash based on file name or contents of file? Can we add a comment either way?